react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronRight, faChevronLeft, faCircle } from '@fortawesome/free-solid-svg-icons';
+import { faCircle as farCircle } from '@fortawesome/free-regular-svg-icons';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { counter: 0, url: '', dotState: faCircle };
+    this.handleDots = this.handleDots.bind(this);
+    this.handleArrowRight = this.handleArrowRight.bind(this);
+    this.handleArrowLeft = this.handleArrowLeft.bind(this);
+    this.change = this.change.bind(this);
+  }
+
+  handleDots(event) {
+    if (event.target === event.currentTarget) {
+      const eventId = Number(event.target.id);
+      this.setState({ counter: eventId, url: this.props.img[eventId] });
+    }
+  }
+
+  handleArrowRight(event) {
+    if (event.target === event.currentTarget && this.state.counter < 3) {
```

Comment:
Using the number `3` here will limit the size of image array that can be passed to your carousel. The carousel should work with any size of image array. Instead of hardcoding `3` here what else could you use to check if you reached the end of the array?